src/dataview.ts

```diff
@@ -390,30 +376,7 @@ export class DataView {
       this.updated = {}
     }
     this.updated[id] = true
-    this.refresh()
-  }
-
-  insertItem(insertBefore: number, item: Item): void {
-    this.items.splice(insertBefore, 0, item)
-    this.updateIdxById(insertBefore)
-    this.refresh()
-  }
-
-  addItem(item: Item): void {
-    this.items.push(item)
-    this.updateIdxById(this.items.length - 1)
-    this.refresh()
-  }
-
-  deleteItem(id: number): void {
-    var idx = this.idxById[id]
-    if (idx === undefined) {
-      throw 'Invalid id'
-    }
-    delete this.idxById[id]
-    this.items.splice(idx, 1)
-    this.updateIdxById(idx)
-    this.refresh()
+    this.refresh({ triggerFilteredItemsChanged: true })
```

Comment:
Arguable, but it may be cleaner to just set `this.triggerFilteredItemsChanged = true` before calling `refresh()`, so `refresh` doesn't have to care about it.